src/libcommon/log/sentry/handler.cpp

```diff
@@ -434,7 +446,158 @@ SentryHandler::~SentryHandler() {
     }
 }
 
-SentryHandler::SentryEvent::SentryEvent(const std::string &title, const std::string &message, SentryLevel level,
+Handler::SentryEvent::SentryEvent(const std::string &title, const std::string &message, Level level,
                                         SentryConfidentialityLevel confidentialityLevel, const SentryUser &user) :
-    title(title), message(message), level(level), confidentialityLevel(confidentialityLevel), userId(user.userId()) {}
-} // namespace KDC
+    title(title),
+    message(message), level(level), confidentialityLevel(confidentialityLevel), userId(user.userId()) {}
+
+void Handler::stopPTrace(const pTraceId &id, bool aborted) {
+    std::scoped_lock lock(_mutex);
+    if (id == 0) return;
+    auto performanceTraceIt = _performanceTraces.find(id);
+    if (performanceTraceIt == _performanceTraces.end()) {
+        return;
+    }
+    const PerformanceTrace &performanceTrace = performanceTraceIt->second;
+
+    // Stop any child PerformanceTrace
+    std::vector<pTraceId> toDelete;
+    for (const auto &childPerformanceTraceId: performanceTrace.children()) {
+        toDelete.push_back(childPerformanceTraceId); // We can't delete directly the childPerformanceTraceId from the map as we
```

Comment:
"We can't delete directly ...": it is usually possible to erase as we go, using an iterator.

Reply:
Thank for pointing it! Done here: [ad7999d](https://github.com/Infomaniak/desktop-kDrive/pull/403/commits/ad7999da434b0ba9d0884007cd538f06d3301cb6)